lib/KBox/src/KMessage.h

```diff
@@ -191,6 +202,92 @@ class IMUMessage: public KMessage {
     };
 };
 
+class NAVMessage: public KMessage { //RIGM added for autopilot functionality
+private:
+    String label; //not sure we need a label
+    bool apMode, apHeadingMode, apWaypointMode, apDodgeMode;
+    double currentHeading, targetHeading, courseToWaypoint, rudderSensorPosition, targetRudderPosition, rudderCommandSent;
+    
+public:
+    NAVMessage(String label, bool apMode, bool apHeadingMode, bool apWaypointMode, bool apDodgeMode, double currentHeading, double targetHeading, double courseToWaypoint, double rudderSensorPosition) :
+    label(label), apMode(apMode), apHeadingMode(apHeadingMode), apWaypointMode(apWaypointMode), apDodgeMode(apDodgeMode),
+    currentHeading(currentHeading),targetHeading(targetHeading),courseToWaypoint(courseToWaypoint),
+    rudderSensorPosition(rudderSensorPosition){};
+    
+    void accept(KVisitor &v) const {
+        v.visit(*this);
+    };
+    
+    String getLabel() const {
+        return label;
+    };
+    
+    bool getApMode() const {
+        return apMode;
+    };
+    
+    bool getApHeadingMode() const {
+        return apHeadingMode;
+    };
+    
+    bool getApWaypointMode() const {
+        return apWaypointMode;
+    };
+    
+    bool getApDodgeMode() const {
+        return apDodgeMode;
+    };
+    
+    /*
+     * Courses in Degrees, therefore using doubles:
```

Comment:
Will note that headings are magnetic.
With respect to the data passed back and forth between the nav page and the autopilot task, yes we could have the Kmessage heading-related variables in radians, but assuming we are doing the autopilot math in degrees, that would mean converting degrees to radians in the navPage and then immediately converting degrees to radians in the autopilot task.  Seems that this would add calculation overhead for no benefit.  On the other hand, the IMU Message is in radians, and this makes it easier to transmit directly to N2K.